hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CachePutAllBackupOperation.java

```diff
@@ -68,11 +75,25 @@ protected void writeInternal(ObjectDataOutput out) throws IOException {
         out.writeBoolean(cacheRecords != null);
         if (cacheRecords != null) {
             out.writeInt(cacheRecords.size());
-            for (Map.Entry<Data, CacheRecord> entry : cacheRecords.entrySet()) {
+            BitSet nonWanReplicatedKeys = new BitSet(cacheRecords.size());
+            int index = 0;
+            for (Map.Entry<Data, WanWrappedCacheRecord> entry : cacheRecords.entrySet()) {
                 Data key = entry.getKey();
-                CacheRecord record = entry.getValue();
+                WanWrappedCacheRecord wrapped = entry.getValue();
                 IOUtil.writeData(out, key);
-                out.writeObject(record);
+                out.writeObject(wrapped.getRecord());
+
+                if (!wrapped.isWanReplicated()) {
+                    nonWanReplicatedKeys.set(index);
+                }
+                index++;
+            }
+
+            if (nonWanReplicatedKeys.isEmpty()) {
```

Comment:
conditionally serialize `if (out.getVersion().isGreaterOrEqual(V5_4))`

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/24928/commits/12f00318cd7d78ba56a65173931efd2a5d621580